src/types/multimin.rs

```diff
@@ -0,0 +1,297 @@
+            n : x.len(),
+            params: &f as *const _ as *mut _,
+        };
+ 
+        self.inner_closure = Some(Box::new(f));
+
+        let ret = unsafe {
+            sys::gsl_multimin_fminimizer_set(
+                self.unwrap_unique(),
+                &mut self.inner_call,
+                x.unwrap_shared(),
+                step_size.unwrap_shared(),
+            )
+        };
+        result_handler!(ret, ())
+    }
+
+	/// This function returns a pointer to the name of the minimizer.
+    #[doc(alias = "gsl_multimin_fminimizer_name")]
+    pub fn name(&self) -> Option<String> {
+        let n = unsafe { sys::gsl_multimin_fminimizer_name(self.unwrap_shared()) };
+        if n.is_null() {
+            return None;
+        }
+        let mut len = 0;
+        loop {
+            if unsafe { *n.offset(len) } == 0 {
+                break;
+            }
+            len += 1;
+        }
+        let slice = unsafe { std::slice::from_raw_parts(n as _, len as _) };
+        std::str::from_utf8(slice).ok().map(|x| x.to_owned())
+    }
+
+    #[doc(alias = "gsl_multimin_fminimizer_x")]
+    pub fn x(&self) -> View<'_,VectorF64> {
+        unsafe { View::new(sys::gsl_multimin_fminimizer_x(self.unwrap_shared())) }
+    }
+
+    #[doc(alias = "gsl_multimin_fminimizer_minimum")]
+    pub fn minimum(&self) -> f64 {
+        unsafe { sys::gsl_multimin_fminimizer_minimum(self.unwrap_shared()) }
+    }
+
+    #[doc(alias = "gsl_multimin_fminimizer_size")]
+    pub fn size(&self) -> f64 {
+        unsafe { sys::gsl_multimin_fminimizer_size(self.unwrap_shared()) }
+    }
+
+	/// This function performs a single iteration of the minimizer s. If the iteration encounters 
+	/// an unexpected problem then an error code will be returned. The error code `Value::NoProgress` 
+	/// signifies that the minimizer is unable to improve on its current estimate, either due 
+	/// to numerical difficulty or because a genuine local minimum has been reached.
+    #[doc(alias = "gsl_multimin_fminimizer_iterate")]
+    pub fn iterate(&mut self) -> Result<(), Value> {
+        let ret = unsafe { sys::gsl_multimin_fminimizer_iterate(self.unwrap_unique()) };
+        result_handler!(ret, ())
+    }
+}
+
+ffi_wrapper!(MinimizerType, *const sys::gsl_multimin_fminimizer_type);
+
+impl MinimizerType {
+    #[doc(alias = "gsl_multimin_fminimizer_nmsimplex2")]
+    pub fn nm_simplex2() -> Self {
+        ffi_wrap!(gsl_multimin_fminimizer_nmsimplex2)
+    }
+
+    #[doc(alias = "gsl_multimin_fminimizer_nmsimplex")]
+    pub fn nm_simplex() -> Self {
+        ffi_wrap!(gsl_multimin_fminimizer_nmsimplex)
+    }
+
+    #[doc(alias = "gsl_multimin_fminimizer_nmsimplex2rand")]
+    pub fn nm_simplex2_rand() -> Self {
+        ffi_wrap!(gsl_multimin_fminimizer_nmsimplex2rand)
+    }
+}
+
+#[cfg(any(test, doctest))]
+mod test {
+    /// This doc block will be used to ensure that the closure can't be set everywhere!
+    ///
+    /// ```compile_fail
+	/// extern crate rgsl;
```

Comment:
I wonder though, did your IDE format like this? :laughing:

Reply:
yes I was mixturing tabs and spaces! I remove all tabs.